src/extension.ts

```diff
@@ -1044,32 +1030,30 @@ export class ExtensionManager implements vscode.Disposable {
         return 0;
     }
 
-    private getProjectFromFolder(folder?: vscode.WorkspaceFolder | string) {
+    private getProjectFromFolder(folder?: vscode.WorkspaceFolder | string, sourceDir?: string) {
         const workspaceFolder: vscode.WorkspaceFolder | undefined = this.getWorkspaceFolder(folder);
         if (workspaceFolder) {
             const activeProject: CMakeProject | undefined = this.getActiveProject();
             const projects: CMakeProject[] | undefined = this.projectController.getProjectsForWorkspaceFolder(workspaceFolder);
             if (!projects || projects.length === 0) {
                 return activeProject;
             } else {
-                for (const project of projects) {
-                    // Choose the active project.
-                    if (activeProject?.folderPath === project.folderPath) {
-                        return project;
-                    }
-                }
-                // Choose the first project folder.
-                return projects[0];
+                // Choose project by corresponding source directory
+                return projects.find(project => sourceDir && (path.normalize(sourceDir) === path.normalize(project.folderPath))) ??
```

Comment:
Assuming this is all one large return statement, could you correct the indentation?

Reply:
```
                // Choose project by corresponding source directory
                return projects.find(project => sourceDir && (path.normalize(sourceDir) === path.normalize(project.folderPath)))
                    // Choose project by folder of active project
                    ?? projects.find(project => activeProject?.folderPath === project.folderPath)
                    // Fallback to first project
                    ?? projects[0];
```
Can I indent this statement that way?